consensus/e2e_tests/state_sync_test.go

```diff
@@ -0,0 +1,161 @@
+package e2e_tests
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/shared/codec"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+func TestStateSyncServerGetMetaDataReq(t *testing.T) {
+	// Test preparation
+	clockMock := clock.NewMock()
+	timeReminder(t, clockMock, time.Second)
+
+	runtimeMgrs := GenerateNodeRuntimeMgrs(t, numValidators, clockMock)
+	buses := GenerateBuses(t, runtimeMgrs)
+
+	// Create & start test pocket nodes
+	eventsChannel := make(modules.EventsChannel, 100)
+	pocketNodes := CreateTestConsensusPocketNodes(t, buses, eventsChannel)
+	StartAllTestPocketNodes(t, pocketNodes)
+
+	testHeight := uint64(4)
+
+	// Choose node 1 as the server node, enable server mode of the node 1
+	// Set server node's height to test height.
+	serverNode := pocketNodes[1]
+	serverNodeConsensusModImpl := GetConsensusModImpl(serverNode)
+	serverNodeConsensusModImpl.MethodByName("SetHeight").Call([]reflect.Value{reflect.ValueOf(testHeight)})
+	serverNodeConsensusModImpl.MethodByName("EnableServerMode").Call([]reflect.Value{})
+
+	// We choose node 2 as the requester node.
+	requesterNode := pocketNodes[2]
+	requesterNodePeerId, err := requesterNode.GetBus().GetConsensusModule().GetCurrentNodeAddressFromNodeId()
+	require.NoError(t, err)
+
+	// Test MetaData Req
+	stateSyncMetaDataReq := typesCons.StateSyncMetadataRequest{
+		PeerId: requesterNodePeerId,
+	}
+
+	stateSyncMetaDataReqMessage := &typesCons.StateSyncMessage{
+		MsgType: typesCons.StateSyncMessageType_STATE_SYNC_METADATA_REQUEST,
+		Message: &typesCons.StateSyncMessage_MetadataReq{
+			MetadataReq: &stateSyncMetaDataReq,
+		},
+	}
+	anyProto, err := anypb.New(stateSyncMetaDataReqMessage)
+	require.NoError(t, err)
+
+	// Send metadata request to the server node
+	P2PSend(t, serverNode, anyProto)
+
+	// Start waiting for the metadata request on server node,
+	receivedMsg, err := WaitForNetworkStateSyncEvents(t, clockMock, eventsChannel, typesCons.StateSyncMessageType_STATE_SYNC_METADATA_RESPONSE, 1, 250, false)
+	require.NoError(t, err)
```

Comment:
Can we add a negative test case as well?

Reply:
Normal peer should reply with its metadata, and even if it lies about it we can't understand if it is lying or not. I think, only way of failing test here would be peer not replying before timeout.

I will think about this, and comment if I find some meaningful way.